src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/VisitSchedulerService.kt

```diff
@@ -222,9 +224,15 @@ class VisitSchedulerService(
   private fun mapVisitDtoToOrchestrationVisitDto(visits: List<VisitDto>?): List<OrchestrationVisitDto> {
     val prisonerIds = visits?.map { it.prisonerId }?.toSet() ?: emptySet()
     val prisonerContactsMap = prisonerContactService.getPrisonersContacts(prisonerIds)
-    return visits?.map {
-      val contacts = prisonerContactsMap[it.prisonerId] ?: emptyList()
-      orchestrationVisitDtoBuilder.build(it, contacts)
+    return visits?.map { visit ->
+      val contacts = prisonerContactsMap[visit.prisonerId] ?: emptyList()
+      try {
+        val prisoner = prisonerSearchService.getPrisoner(visit.prisonerId)
```

Comment:
could we do this once outside the loop to reduce calls and add to a Map?

Reply:
Great suggestion, as discussed on slack - fixed this issue